mteb/evaluation/evaluators/RetrievalEvaluator.py

```diff
@@ -561,8 +561,13 @@ def evaluate(
             recall[f"Recall@{k}"] = round(sum(recall[f"Recall@{k}"]) / len(scores), 5)
             precision[f"P@{k}"] = round(sum(precision[f"P@{k}"]) / len(scores), 5)
 
+        # results needs to be only those with qrels
+        abstention_results = {
+            key: value for key, value in scores.items() if key in qrels
+        }
         naucs = RetrievalEvaluator.evaluate_abstention(
-            results, {**all_ndcgs, **all_aps, **all_recalls, **all_precisions}
+            abstention_results,
```

Comment:
I have tested that we get reasonable scores and that evaluation works on all the datasets.  Two of them gave me a very weird error: NeuCLIR2022 and MIRACL.  For both of these they get an error in the `evaluate_abstention` function about an out of 

NeuCLIR was solved by adding this fix. The `scores` dict has as many qids as the `qrels` but somehow the `results` had more. This is still failing for MIRACL though so I am trying to debug now. 

Unfortunately I have a workshop I am attending today so I will be out until the evening. I'm hoping to solve this in the next hour but just as an FYI

Reply:
Okay nevermind.  The NeuCLIR2022 error is from this (but I think this changes the abstention results in a negative way for other tasks).  The MIRACL ones seems to be coming from a different place, I haven't localized it yet. 

I'll be back around 8pm PT and will work on it or if either of you has time you can take a look.  I'm just running a weak model on it:

```
from mteb import MTEB
from sentence_transformers import SentenceTransformer
import logging

# Set up logging
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

# Define the model to use
model_name = "intfloat/multilingual-e5-small"

# Define the tasks
tasks = [
    "NeuCLIR2022Retrieval" , "MIRACLRetrieval"
]

def run_task(task_name, model):
    task_name_fast = f"{task_name}HardNegatives"
    logger.info(f"Running task: {task_name_fast}")
    try:
        task = MTEB([task_name_fast])
        results = task.run(model, output_folder=f"results/{task_name_fast}")
        logger.info(f"Task {task_name_fast} completed successfully")
        return results
    except Exception as e:
        logger.error(f"Error running task {task_name_fast}: {e}")
        return None

def main():
    # Load the model
    model = SentenceTransformer(model_name)

    for task in tasks:
        results = run_task(task, model)
        if results:
            logger.info(f"Results for {task}-Fast: {results}")

if __name__ == "__main__":
    main()
```

Reply:
We could also merge and run all tasks except MIRACL and NeuCLIR2022, although MIRACL is pretty important....

Reply:
I figured it out, re-running now. Will update tomorrow morning